src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -152,5 +165,25 @@ public virtual IValue ToAST(object value)
                 return null;
             }
         }
+
+        protected internal object ThrowLiteralConversionError(IValue input)
+        {
+            throw new ArgumentException($"Unable to convert '{input}' to '{Name}'");
+        }
+
+        protected internal IValue ThrowASTConversionError(object value)
+        {
+            throw new InvalidOperationException($"Unable to convert '{value}' of the scalar type '{Name}' to an AST representation.");
+        }
+
+        protected internal object ThrowValueConversionError(object value)
+        {
+            throw new ArgumentException($"Unable to convert '{value}' to '{Name}'");
+        }
+
+        protected internal object ThrowSerializationError(object value)
```

Comment:
Used only fro enums?

Reply:
We will see.  Right now, `ThrowValueConversionError` isn't used at all!  But that will change once `ValueConverter` is removed. 
 And since the default implementation of `Serialize` calls `ParseValue`, we may get exception messages intended for `ParseValue` be returned during a failed call to `Serialize`.  Something we need to think about during the review.  Still a WIP just now.

Reply:
I'm OK with current state.